Composer/packages/lib/shared/src/copyUtils/copyForeach.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Foreach, ForeachPage } from '../types';
+
+import { ExternalApi } from './ExternalApi';
+import { shallowCopyAdaptiveAction } from './shallowCopyAdaptiveAction';
+import { copyAdaptiveActionList } from './copyAdaptiveActionList';
+
+type ForeachAction = Foreach | ForeachPage;
+export const copyForeach = async (input: ForeachAction, externalApi: ExternalApi): Promise<ForeachAction> => {
+  const copy = shallowCopyAdaptiveAction(input, externalApi);
+
+  if (Array.isArray(input.actions)) {
```

Comment:
is it possible to remove this checking, since you already checked that inside copyList function

Reply:
it's similar to previous comment, `actions` are declared as optional in SDK. If we don't check it here, the copy result will be different from input value (`[]` against `undefined`)